library/src/test/java/com/novoda/downloadmanager/lib/QueryTest.java

```diff
@@ -48,4 +49,39 @@ public void givenNoBatchIdsWhenTheQueryIsCreatedThenTheWhereStatementContainsNoB
 
         assertThat(stringArgumentCaptor.getValue()).doesNotContain(Downloads.Impl.COLUMN_BATCH_ID + " IN ");
     }
+
+    @Test
+    public void whenWeSetABatchStatusOrderByOnAQueryThenTheResolverIsQueriedWithTheCorrectSortOrder() {
+        new Query().orderBy(DownloadManager.COLUMN_LAST_MODIFIED_TIMESTAMP, Query.ORDER_ASCENDING).runQuery(resolver, null, uri);
+
+        // Actually resolves to Downloads.Impl.COLUMN_LAST_MODIFIED
+        verify(resolver).query(any(Uri.class), any(String[].class), anyString(), any(String[].class), eq("lastmod ASC"));
+    }
+
+    @Test
+    public void whenWeSetNoOrderByOnAQueryThenTheResolverIsQueriedWithTheLastModifiedSortOrder() {
+        new Query().runQuery(resolver, null, uri);
+
+        verify(resolver).query(any(Uri.class), any(String[].class), anyString(), any(String[].class), eq("lastmod DESC"));
+    }
+
+    @Test
+    public void whenOrderingByLivenessThenTheResolverIsQueriedWithTheExpectedSort() {
+        new Query().orderByLiveness().runQuery(resolver, null, uri);
+
+        verify(resolver).query(any(Uri.class), any(String[].class), anyString(), any(String[].class), eq("CASE batch_status " +
+                        "WHEN 192 THEN 1 " +
+                        "WHEN 190 THEN 2 " +
+                        "WHEN 193 THEN 3 " +
+                        "WHEN 498 THEN 4 " +
+                        "WHEN 200 THEN 5 " +
+                        "ELSE 2 END"));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenWeSetAnUnsupportedOrderByOnAQueryThenTheResolverIsQueriedWithTheCorrectSortOrder() {
+        new Query().orderBy(DownloadManager.COLUMN_STATUS, Query.ORDER_ASCENDING).runQuery(resolver, null, uri);
```

Comment:
would be nice to pull out `Query.ORDER_ASCENDING` to a var explaining it's unsupported